engine/engine.go

```diff
@@ -241,24 +238,26 @@ func (e *EngineAPI) GetPayloadV3(payloadID engine.PayloadID) (*eth.ExecutionPayl
 	e.lock.RLock()
 	defer e.lock.RUnlock()
 
-	payload := e.payloadStore.Current()
-	if currentID := payload.ID(); payloadID != *currentID {
+	if payloadID != *e.payloadAttr.ID() {
```

Comment:
I think explicitly checking nil is fine here. I agree the safest approach is to check for nil internally in general, but because the engine is the one setting the value to nil, I believe it bears responsibility here.